CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+## 3.16.0 (June 7, 2023)
+FEATURES:
+* VAULT-8099 LDAP secrets engine: ([#1859](https://github.com/hashicorp/terraform-provider-vault/pull/1859))
+* Data source vault auth backends: ([#1827](https://github.com/hashicorp/terraform-provider-vault/pull/1827))
+* Support allowed_domains_template on ssh_secret_backend_role. Fixes hashicorp#1675: ([#1676](https://github.com/hashicorp/terraform-provider-vault/pull/1676))
+
+IMPROVEMENTS:
+* Add support for retrying kv-v2 secret data writes: ([#1887](https://github.com/hashicorp/terraform-provider-vault/pull/1887))
+* Add back support for deriving the provider namespace from the Vault token's: ([#1841](https://github.com/hashicorp/terraform-provider-vault/pull/1841))
+
+BUGS:
+* check version for allowed_domains_template field: ([#1881](https://github.com/hashicorp/terraform-provider-vault/pull/1881))
```

Comment:
nit: Wondering if we can remove this, this was a fix for an unreleased change, so not sure if it warrants being on the changelog

Reply:
Yep, totally agree. Will update